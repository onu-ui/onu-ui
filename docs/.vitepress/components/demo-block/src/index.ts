import type { ExtractPropTypes } from 'vue'

export const demoProps = {
  github: {
    type: String,
    default: 'https://github.com/onu-ui/onu-ui',
  },
  codeSandBox: {
    type: String,
    default: 'https://codesandbox.io/',
  },
  highlightedCode: {
    type: String,
    default: '',
  },
  code: {
    type: String,
    default: '',
  },
  title: {
    type: String,
    default: '',
  },
  desc: {
    type: String,
    default: '',
  },
  lang: {
    type: String,
    default: 'vue',
  },
  expand: {
    type: Boolean,
    default: false,
  },
} as const

export type ODemoProps = ExtractPropTypes<typeof demoProps>
