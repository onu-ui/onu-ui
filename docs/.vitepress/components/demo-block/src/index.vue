<script lang='ts' setup name="demo-block">
import { demoProps } from './index'

const props = defineProps(demoProps)

const decodedHighlightedCode = computed(() =>
  decodeURIComponent(props.highlightedCode),
)
const { copy, copied } = useClipboard({ source: decodeURIComponent(props.code) })
const [value, toggle] = useToggle()
</script>

<template>
  <ClientOnly>
    <div v-bind="$attrs" class="mt-6">
      <div class="o-demo_wrapper vp-raw bg">
        <slot />
      </div>
      <div class="relative">
        <div class="o-demo_actions">
          <a class="o-demo_action_item" group :href="codeSandBox" target="_blank">
            <div class="o-demo_action_icon i-carbon:chemistry" />
            <div class="o-demo_tooltip" group-hover:opacity-100>
              Open on CodeSandBox
            </div>
          </a>
          <a class="o-demo_action_item" group :href="github" target="_blank">
            <div class="o-demo_action_icon i-carbon-logo-github" />
            <div class="o-demo_tooltip" group-hover:opacity-100>
              Edit on GitHub
            </div>
          </a>
          <a class="o-demo_action_item" group @click="copy()">
            <div class="o-demo_action_icon i-carbon:copy" />
            <div class="o-demo_tooltip" group-hover:opacity-100>
              {{ copied ? 'Copied' : 'Copy code' }}
            </div>
          </a>
          <a class="o-demo_action_item" group @click="toggle()">
            <div class="o-demo_action_icon i-carbon:fit-to-width" />
            <div class="o-demo_tooltip" group-hover:opacity-100>
              {{ value ? 'Hidden code' : 'Show code' }}
            </div>
          </a>
        </div>
        <div v-show="value" :class="`language-${lang} extra-class`" v-html="decodedHighlightedCode" />
      </div>
    </div>
  </ClientOnly>
</template>

<!-- <style scoped>
.bg {
      /* background-color: #1c1f24; */
      background: conic-gradient(from 192deg at 46.5% 42.58%, #FA8792 -54.94deg, #561BBE 28.29deg,
          #70E6FB 157.82deg, #FBF8B3 220.83deg, #FA8792 305.06deg, #561BBE 388.29deg);
    }
</style> -->
