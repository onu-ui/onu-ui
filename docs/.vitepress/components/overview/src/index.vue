<script lang="ts" setup>
import { Components } from '@onu-ui/docs/guide/components'
import SubTitle from './SubTitle.vue'
import ComponentCard from './ComponentCard.vue'
</script>

<template>
  <div class="space-y-10">
    <div v-for="(category, cKey) in Components" :key="cKey">
      <SubTitle>
        {{ category.name }}
        <span text-xl>{{ `(${category.components.length})` || '' }}</span>
      </SubTitle>
      <div class="grid md:grid-cols-3  gap-5">
        <ComponentCard v-for="(component, pKey) in category.components" :key="pKey" :component="component" />
      </div>
    </div>
  </div>
</template>
