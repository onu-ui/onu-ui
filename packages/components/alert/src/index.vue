<script lang='ts' setup name="O-Alert">
import OIcon from '../../icon/src/index.vue'
import { alertEmits, alertProps } from './props'

const props = defineProps(alertProps)
const emit = defineEmits(alertEmits)
const slots = useSlots()

const visible = ref(true)

const handleClose = (evt: MouseEvent) => {
  visible.value = false
  emit('close', evt)
}

const showIcon = computed(() => props.icon || slots.icon)
const showTitle = computed(() => props.title || slots.title)
</script>

<template>
  <div v-show="visible" :o="o" class="o-alert-base" :class="[light && 'o-alert-light']">
    <!-- icon -->
    <div v-if="showIcon" class="o-alert-icon">
      <slot name="icon">
        <o-icon :name="icon" class="!c-white text-6" />
      </slot>
    </div>
    <!-- content -->
    <div class="o-alert-content" :class="center && 'items-center'">
      <div class="o-alert-header">
        <div v-if="showTitle" class="o-alert-title">
          <slot name="title">
            {{ title }}
          </slot>
        </div>
        <div v-if="closable" class="o-alert-close" @click="handleClose">
          <slot name="close">
            <o-icon v-if="!closeText" name="i-carbon-close" class="!c-white text-4" />
            <span class="text-xs">{{ closeText }}</span>
          </slot>
        </div>
      </div>
      <div class="o-alert-description" :class="showTitle ? 'mt-1' : ''">
        <slot>
          {{ description }}
        </slot>
      </div>
    </div>
  </div>
</template>
