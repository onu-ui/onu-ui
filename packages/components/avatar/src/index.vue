<script lang='ts' setup name="OAvatar">
import { avatarProps } from './props'

const props = defineProps(avatarProps)

const IsBgMode = computed(() => props.mode === 'bg')
</script>

<template>
  <component
    :is="IsBgMode ? 'div' : 'img'"
    class="o-transition o-avatar-base"
    :class="[IsBgMode ? 'o-avatar-mode-bg' : 'o-avatar-mode-img', `o-avatar-${size}`]"
    v-bind="IsBgMode ? { style: { backgroundImage: `url(${src})` } } : { src, alt: 'Avatar' }"
  >
    <slot v-if="IsBgMode" />
  </component>
</template>
