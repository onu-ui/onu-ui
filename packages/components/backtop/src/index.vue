<script setup lang="ts" name="OBacktop">
import { backtopEmits, backtopProps } from './backtop'
import { useBackTop } from './use-backtop'

const props = defineProps(backtopProps)
const emit = defineEmits(backtopEmits)

const { handleClick, visible } = useBackTop(props, emit, 'OBacktop')
const backTopStyle = computed(() => ({
  right: `${props.right}px`,
  bottom: `${props.bottom}px`,
}))
</script>

<template>
  <transition>
    <div
      v-if="visible"
      :style="backTopStyle"
      class="fixed bg-blue-500 shadow-lg shadow-context:50 w-40px h-40px rounded-1/2 flex justify-center items-center text-20px cursor-pointer z-5 hover:bg-blue-400"
      @click.stop="handleClick"
    >
      <slot>
        <div i-carbon-up-to-top class="!c-white" />
      </slot>
    </div>
  </transition>
</template>

<style scoped>
.v-enter-active,
.v-leave-active {
  transition: opacity 0.5s ease;
}

.v-enter-from,
.v-leave-to {
  opacity: 0;
}
</style>
