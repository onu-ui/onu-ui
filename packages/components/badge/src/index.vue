<script lang="ts" setup name="OBadge">
import { isNumber } from '@onu-ui/utils'
import { badgeProps } from './props'

const props = defineProps(badgeProps)

const resolveValue = computed(() =>
  props.dot
    ? ''
    : (isNumber(props.value) && props.value > props.max)
        ? `${props.max}+`
        : props.value,
)
</script>

<template>
  <span class="o-badge" o-error v-bind="$attrs">
    <slot />
    <sup
      v-if="resolveValue !== 0 || showZero"
      class="o-badge-sup o-badge-sup-fixed"
      :class="dot && 'o-badge-sup-dot'"
    >{{ resolveValue }}</sup>
  </span>
</template>
