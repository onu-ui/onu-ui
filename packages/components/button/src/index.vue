<script setup lang="ts" name="O-Button">
 interface IButtonProps {
  to?: string
  icon?: string
  light?: boolean
}

withDefaults(defineProps<IButtonProps>(), {
  to: '',
  icon: 'medium',
})
</script>

<template>
  <component
    :is="to ? 'a' : 'button'"
    v-bind="to ? { href: to } : {}"
    class="o-transition o-button-base"
    :class="light ? 'o-button-light' : ''"
  >
    <slot name="icon">
      <!-- <NIcon v-if="icon" :icon="icon" class="c-buttoc-icon" /> -->
    </slot>
    <slot />
  </component>
</template>
