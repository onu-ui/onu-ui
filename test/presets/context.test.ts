import { describe, expect, it } from 'vitest'
import { createGenerator } from 'unocss'
import { presetOnu } from '../../packages/preset/src/index'
import { contextTargets, unContextTargets } from '../features/assets/context-targets'

describe('onu context', () => {
  const testColor = '#9955FF'

  const uno = createGenerator({
    presets: [presetOnu({ color: testColor })],
  })

  it('matched tokens', async () => {
    const { css, matched } = await uno.generate(contextTargets, { preflights: false })
    expect(css).toMatchSnapshot()
    expect(matched.size).toBe(contextTargets.length)
  })

  it('unmatched tokens', async () => {
    const { matched } = await uno.generate(unContextTargets, { preflights: false })
    expect(matched.size).toBe(0)
  })
})
