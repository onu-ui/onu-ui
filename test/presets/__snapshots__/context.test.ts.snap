// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`onu context > matched tokens 1`] = `
"/* layer: default */
.o-\\[rgb\\(255\\,255\\,0\\)\\],
.o-\\#ff0{--onu-color-context:60 100 50;}
.o-red{--onu-color-context:0 91 71;}
.o-theme-100{--onu-color-context:var(--onu-color-100);}
.o-theme-bg{--onu-color-context:var(--onu-color-bg);}
.o-theme-border{--onu-color-context:var(--onu-color-border);}
.o-theme-DEFAULT{--onu-color-context:var(--onu-color-DEFAULT);}
.o-bg-\\[rgb\\(255\\,255\\,0\\)\\],
.o-bg-\\#ff0{--onu-color-bg:60 100 50;}
.o-bg-red{--onu-color-bg:0 91 71;}
.o-bg-theme-100{--onu-color-bg:var(--onu-color-100);}
.o-bg-theme-DEFAULT{--onu-color-bg:var(--onu-color-DEFAULT);}
.o-text-\\[rgb\\(255\\,255\\,0\\)\\],
.o-text-\\#ff0{--onu-color-text:60 100 50;}
.o-text-red{--onu-color-text:0 91 71;}
.o-text-theme-100{--onu-color-text:var(--onu-color-100);}
.o-text-theme-DEFAULT{--onu-color-text:var(--onu-color-DEFAULT);}
.o-border-\\[rgb\\(255\\,255\\,0\\)\\],
.o-border-\\#ff0{--onu-color-border:60 100 50;}
.o-border-red{--onu-color-border:0 91 71;}
.o-border-theme-100{--onu-color-border:var(--onu-color-100);}
.o-border-theme-DEFAULT{--onu-color-border:var(--onu-color-DEFAULT);}
.bg-theme-100{--un-bg-opacity:1;background-color:hsl(var(--onu-color-100, var(--onu-color-context)) / var(--un-bg-opacity));}
.bg-theme-100-50,
.bg-theme-100\\:50{--un-bg-opacity:0.5;background-color:hsl(var(--onu-color-100, var(--onu-color-context)) / var(--un-bg-opacity));}
.bg-theme-context{--un-bg-opacity:1;background-color:hsl(var(--onu-color-bg, var(--onu-color-context)) / var(--un-bg-opacity));}
.bg-theme-DEFAULT{--un-bg-opacity:1;background-color:hsl(var(--onu-color-DEFAULT, var(--onu-color-context)) / var(--un-bg-opacity));}
.text-theme-100{--un-text-opacity:1;color:hsl(var(--onu-color-100, var(--onu-color-context)) / var(--un-text-opacity));}
.text-theme-100-50,
.text-theme-100\\:50{--un-text-opacity:0.5;color:hsl(var(--onu-color-100, var(--onu-color-context)) / var(--un-text-opacity));}
.text-theme-context{--un-text-opacity:1;color:hsl(var(--onu-color-text, var(--onu-color-context)) / var(--un-text-opacity));}
.text-theme-DEFAULT{--un-text-opacity:1;color:hsl(var(--onu-color-DEFAULT, var(--onu-color-context)) / var(--un-text-opacity));}
.border-theme-100{--un-border-opacity:1;border-color:hsl(var(--onu-color-100, var(--onu-color-context)) / var(--un-border-opacity));}
.border-theme-100-50,
.border-theme-100\\:50{--un-border-opacity:0.5;border-color:hsl(var(--onu-color-100, var(--onu-color-context)) / var(--un-border-opacity));}
.border-theme-context{--un-border-opacity:1;border-color:hsl(var(--onu-color-border, var(--onu-color-context)) / var(--un-border-opacity));}
.border-theme-DEFAULT{--un-border-opacity:1;border-color:hsl(var(--onu-color-DEFAULT, var(--onu-color-context)) / var(--un-border-opacity));}
.dark .o-theme-text,
.o-theme-text{--un-text-opacity:1;color:hsl(var(--onu-color-text) / var(--un-text-opacity));}"
`;
